rfcs/text/0030-linux-event-model.md

```diff
@@ -0,0 +1,1299 @@
+-->
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+<!--
+Stage 2: Add or update all remaining field definitions. The list should now be exhaustive. The goal here is to validate the technical details of all remaining fields and to provide a basis for releasing these field definitions as beta in the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
+-->
+
+## Usage
+
+<!--
+Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
+-->
+
+1. Session View
+
+The primary use case for this data will be to drive the "Session view". A time ordered rendering of a linux session process tree. The addition of nested process ancestors entry_leader, session_leader, and group_leader will allow Session View to load only the events for a particular session.
+
+KQL examples:
+process.entry_leader.entity_id: &lt;entity_id of entry session leader>
+
+or
+
+process.session_leader.entity_id: &lt;entity_id of session leader>
+
+group_leader process info as well as file descriptors (process.fds) will allow the Session View to properly represent pipes and redirects in a familiar (shell) way.
+
+observer.boot.id and observer.pid_ns_ino will be used in generating unique uuids for the process using fn(process.pid, process.start, observer.pid_ns_ino, observer.boot.id)
+
+process.tty will be used to determine if the session is interactive. If the field is unset there is no controlling tty and the session is non-interactive (possibly a service).
+
+file_description fieldset will primarily be used to render pipes and redirects, but shed light on processes file and socket activities.
+
+2. Rule engines
+
+The new nested ancestor processes will provide rule engines a widened context to allow for improved specificity/scope. Rather than rely on sequences of events to decide when to take action, an informed decision can be made on a single event. This opens up the door for much quicker and pre-emptive actions. Like stopping of processes, or isolating hosts.
```

Comment:
> This opens up the door for much quicker and pre-emptive actions.

How is this true? Even if each event has more context in contained within itself, that additional context would come from activity previously collected from earlier events. Wouldn't a stateful rule engine like EQL have access to the all the same context at the same time as the event context enrichment phase?

Reply:
Good point, this needs some rewording. It doesn't allow faster response since the last event in the 'sequence' triggers it in both case (whether from a sequence or collected in context).

I think the intent is to highlight EQL on the serverside tends to only see the events in scope of the rule query (e.g. last 5 min) whereas the fatter events have a fuller history.

Reply:
Open to suggestions on a re-wording for this, or I can just remove that sentence all together.